### PR TITLE
refactor(cheatsheet): replaced the cheatsheet api hook with react query

### PR DESCRIPTION
**Pull request by Her0elt**
## replaced the cheatsheet api hook with react query.

Comments/issues/screenshots:
wondring if we should block the site for any one that isnt logged inn or if it is enough to keep it as it is, since you cant GET from backend if you arent a logged inn member

## Pull request checklist

Please check if your PR fulfills the following requirements:
- [ ] The PR includes a `closes #issueID`
- [ ] The PR includes a picture showing visual changes

## Typescript convert checklist

- [ ] All relative imports has been converted to absolute imports
- [ ] All HOC's have been replaced with Hooks
- [ ] Uses theme.spacing() instead of custom padding
- [ ] Uses theme.breakpoints instead of custom media queries

### CODE REVIEW

File: src/containers/Cheatsheet/components/Files.tsx [hunk 1]
```diff
@@ -65,11 +65,12 @@ const useStyles = makeStyles((theme: Theme) => ({
 
 export type FilesProps = {
   files: Array<Cheatsheet>;
-  nextPage: number | null;
-  goToNextPage: () => void;
+  nextPage: boolean | undefined;
```
Comment (olros) [inline src/containers/Cheatsheet/components/Files.tsx:68]: Kan rename denne til `hasNextPage`

File: src/containers/Cheatsheet/index.tsx [hunk 2]
```diff
@@ -39,124 +38,102 @@ const Cheetsheet = () => {
...
   const [input, setInput] = useState('');
   const [search, setSearch] = useState('');
   const [page, setPage] = useState(1);
-  const [nextPage, setNextPage] = useState<number | null>(null);
-  const [files, setFiles] = useState<Array<Cheatsheet>>([]);
-
+  const filters = {
+    page: page,
+    search: search,
+  };
```
Comment (olros) [inline src/containers/Cheatsheet/index.tsx:48]: Flytt denne inn i hooken

File: src/containers/Cheatsheet/index.tsx [hunk 3]
```diff
@@ -39,124 +38,102 @@ const Cheetsheet = () => {
...
+                  navigate(`${URLS.cheatsheet}${studyId}/${newClass}/`);
+                }
+              };
+              
+              useEffect(() => {
+                const timer = setTimeout(() => setSearch(input), 500);
+                return () => clearTimeout(timer);
+              }, [input]);
+              const { data, hasNextPage, fetchNextPage, isLoading} = useCheatsheet(getStudy() || Study.DATAING, getClass() || 1, filters);
+              const files = useMemo(() => (data ? data.pages.map((page) => page.results).flat() : []), [data]);
```
Comment (olros) [inline src/containers/Cheatsheet/index.tsx:136]: Flytt disse oppover i fila rett under `getStudy()` og `getClass()`

File: src/api/hooks/Cheatsheet.tsx [hunk 4]
```diff
@@ -1,8 +1,19 @@
...
-  const getCheatsheets = useCallback((study: Study, grade: number, filters = null) => API.getCheatsheets(study, grade, filters), []);
-  return { getCheatsheets };
+const QUERY_KEY = 'cheatsheet';
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export const useCheatsheet = (study: Study, grade: number, filters?: any) => {
+  return useInfiniteQuery<PaginationResponse<Cheatsheet>, RequestResponse>([QUERY_KEY, study, grade, filters],
+    ({ pageParam = 1 }) => API.getCheatsheets(study, grade, { ...filters, page: pageParam }), 
+    {
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
```
Comment (olros) [inline src/api/hooks/Cheatsheet.tsx:14]: Denne kommentaren bør ikke være nødvendig

**Review by olros — CHANGES_REQUESTED** [review]
Har du formatert med prettier?

**Comment by olros** [comment]
> wondring if we should block the site for any one that isnt logged inn or if it is enough to keep it as it is, since you cant GET from backend if you arent a logged inn member

Hva mener du her? Det jo ikke mulig å komme inn på siden i dag uten å være logget inn?

**Comment by Her0elt** [comment]
> > wondring if we should block the site for any one that isnt logged inn or if it is enough to keep it as it is, since you cant GET from backend if you arent a logged inn member
> 
> Hva mener du her? Det jo ikke mulig å komme inn på siden i dag uten å være logget inn?

ja idk, siden du f[r jo ingen ting n[r du kommer deg did

File: src/containers/Cheatsheet/components/Files.tsx [hunk 5]
```diff
@@ -65,11 +65,12 @@ const useStyles = makeStyles((theme: Theme) => ({
...
   files: Array<Cheatsheet>;
-  nextPage: number | null;
-  goToNextPage: () => void;
+  hasNextPage: boolean | undefined;
+  getNextPage: () => void;
+  isLoading: boolean;
 };
 
-const Files = ({ files, nextPage, goToNextPage }: FilesProps) => {
+const Files = ({ files, hasNextPage: nextPage, getNextPage, isLoading }: FilesProps) => {
```
Comment (olros) [inline src/containers/Cheatsheet/components/Files.tsx:73]: Kan jo endre til hasNextPage på resten av filen også 😉 
```suggestion
const Files = ({ files, hasNextPage, getNextPage, isLoading }: FilesProps) => {
```

File: src/containers/Cheatsheet/index.tsx [hunk 6]
```diff
@@ -39,13 +38,10 @@ const Cheetsheet = () => {
   const classes = useStyles();
   const { studyId, classId } = useParams();
   const navigate = useNavigate();
-  const { getCheatsheets } = useCheatsheet();
   const { getUserData } = useUser();
   const [input, setInput] = useState('');
   const [search, setSearch] = useState('');
   const [page, setPage] = useState(1);
```
Comment (olros) [inline src/containers/Cheatsheet/index.tsx:44]: Med React Query så skal du ikke trenge denne lenger. `fetchNextPage` fikser lasting av neste side

**Review by olros — CHANGES_REQUESTED** [review]

File: src/containers/Cheatsheet/components/Files.tsx (hunk 5 above)
Comment (Her0elt) [reply src/containers/Cheatsheet/components/Files.tsx:73]: haaha f2 gjorde ikke et nei lol

**Review by olros — APPROVED** [review]
Kjør yarn format på `Files.tsx` og så er det bare å merge 💯

**Pull request merged**